packages/aws-cdk-lib/aws-elasticloadbalancingv2/lib/nlb/network-load-balancer.ts

```diff
@@ -220,13 +228,25 @@ export class NetworkLoadBalancer extends BaseLoadBalancer implements INetworkLoa
       type: 'network',
       securityGroups: Lazy.list({ produce: () => this.securityGroups }),
       ipAddressType: props.ipAddressType,
+      enforceSecurityGroupInboundRulesOnPrivateLinkTraffic: Lazy.string({
+        produce: () => this.transformEnforceSecurityGroupInboundRulesOnPrivateLinkTraffic(props.enforceSecurityGroupInboundRulesOnPrivateLinkTraffic),
+      }),
     });
 
     this.metrics = new NetworkLoadBalancerMetrics(this, this.loadBalancerFullName);
     this.isSecurityGroupsPropertyDefined = !!props.securityGroups;
     this.connections = new ec2.Connections({ securityGroups: props.securityGroups });
     this.ipAddressType = props.ipAddressType ?? IpAddressType.IPV4;
     if (props.crossZoneEnabled) { this.setAttribute('load_balancing.cross_zone.enabled', 'true'); }
+    this.enforceSecurityGroupInboundRulesOnPrivateLinkTraffic =
+      this.transformEnforceSecurityGroupInboundRulesOnPrivateLinkTraffic(props.enforceSecurityGroupInboundRulesOnPrivateLinkTraffic);
+  }
+
+  private transformEnforceSecurityGroupInboundRulesOnPrivateLinkTraffic(value: boolean | undefined): string | undefined {
+    if (value !== undefined) {
+      return value ? 'on' : 'off';
+    }
+    return undefined;
```

Comment:
```
  private readonly _enforceSecurityGroupInboundRulesOnPrivateLinkTraffic?: boolean;
  
  ...
  
  public get enforceSecurityGroupInboundRulesOnPrivateLinkTraffic(): string | undefined {
    if (this._enforceSecurityGroupInboundRulesOnPrivateLinkTraffic === undefined) return undefined;
    return this._enforceSecurityGroupInboundRulesOnPrivateLinkTraffic ? 'on' : 'off';
  }
```
What about a private boolean property and a public getter for the transformation?
It seems like a cleaner solution.

Reply:
You would need a setter though, correct, to set `_enforceSecurityGroupInboundRulesOnPrivateLinkTraffic`? How would you set that in `super()`?

Reply:
yeah i'm in a mental 🥨 with these getters/setters

Reply:
🤔 Shouldn't this work with the above getter or am I missing something?
```
  super(scope, id, props, {
      ...
      enforceSecurityGroupInboundRulesOnPrivateLinkTraffic: Lazy.string({
        produce: () => this.enforceSecurityGroupInboundRulesOnPrivateLinkTraffic,
      }),
    });

  this._enforceSecurityGroupInboundRulesOnPrivateLinkTraffic = props.enforceSecurityGroupInboundRulesOnPrivateLinkTraffic
```

Reply:
That's better, thanks man. I was doing the following 👎🏼 -->

```
this.enforceSecurityGroupInboundRulesOnPrivateLinkTraffic = props.enforceSecurityGroupInboundRulesOnPrivateLinkTraffic;
```